go-controller/pkg/ovn/multipolicy_test.go

```diff
@@ -616,8 +624,13 @@ var _ = ginkgo.Describe("OVN MultiNetworkPolicy Operations", func() {
 						gomega.Expect(err).NotTo(gomega.HaveOccurred())
 					}
 
-					ocInfo.asf.EventuallyExpectAddressSetWithAddresses(namespaceName1, []string{"10.1.1.1"})
-
+					if !expectMissingMappingFailure {
+						ocInfo.asf.EventuallyExpectAddressSetWithAddresses(namespaceName1, []string{"10.1.1.1"})
+					} else {
+						ocInfo.asf.EventuallyExpectAddressSetWithAddressesFailure(namespaceName1, []string{"10.1.1.1"})
+						gomega.Expect(fakeOvn.fakeRecorder.Events).To(gomega.HaveLen(1))
```

Comment:
this one is redundant right ? We can have only the one below.

Reply:
it checks the Event was fired, the previous check that there was an error
they are not the same (i.e if we remove the recordError, it can still fail the flow because the mapping wasnt found and the function returns error)
behind the scene, it also helps us, because if the test runs too fast, the event wont manage to be recorded,
so the above line also check Consistently for 2 seconds that there is a failure, which adds the required delay
(reminds the sleep we have between the nad creation and the pod creation in the tests)

Reply:
I think I didn't manage to express myself properly, I apologize for that. 

I mean to say `gomega.Expect(fakeOvn.fakeRecorder.Events).To(gomega.HaveLen(1))` can be removed, and you can leave only `gomega.Expect(<-fakeOvn.fakeRecorder.Events).To(gomega.Equal(`Warning ErrorReconcilingPod failed to find bridge mapping for network: "network1", node "node1", ovn-bridge-mappings: ""`))` . Right ? 

Like ... both of these check the event was fired - one checks the event queue has a single event, the other that we receive the event we're after. I think we can leave only the former.

Reply:
well if there arent event it will block (unless we use select) as far as i know, moreover we want that there will be only one event, else the other might be noise, and also their order can cause noise

Reply:
OK, I give up.

Reply:
Uhm ok, so we are making sure there's only one event and then that it's the event we're expecting. All good.